### PR TITLE
Support multiple `ParameterHandlers` in one `SMIRNOFFCollection`

### PR DESCRIPTION
**Pull request by wwilla7**
### Description

Responds to [Issue#647](https://github.com/openforcefield/openff-interchange/issues/647)

I have left comments in the code block regarding:
1. If more than one collection is loaded by the plugin.
2. If there is only one handler in the collection.

### Checklist

- [ ] Add tests
- [ ] Lint
- [ ] Update docstrings

### CODE REVIEW

File: openff/interchange/smirnoff/_create.py [hunk 1]
```diff
@@ -348,13 +349,30 @@ def _plugins(
...
+    
+    collection = collection_class.create(
+        parameter_handler=handlers,
+        topology=topology,
+    )
+
+    interchange.collections.update(
+        {
+            # handler_class._TAGNAME: collection,
+            collection.type: collection, # Since handlers have different tagnames, I would match collection type instead of handler
```
Comment (mattwthompson) [inline openff/interchange/smirnoff/_create.py]: I think this is a great idea

**Comment by mattwthompson** [comment]
@wwilla7 I made a couple of changes to get this working with the existing tests; I don't think your case should be affected but I don't have it integrated into these tests yet. (Some tests may still fail from your fork since we don't expose the OpenEye license to forks, but if everything here is good I'll move these changes to a separate PR that can access it.) Could you pull down the latest changes and let me know if they work with your plugin?

I pushed directly to your branch, so pulling from your remote to your local clone should bring the changes in. Just make sure 08f9e3b37acf03f0bbe75d10a077cc4055e1a2c9 is the most recent commit in your `git log` and either re-install Interchange or ensure you were using an editable install (i.e. `pip install -e .`).

**Comment by wwilla7** [comment]
Hi, I pulled and tested it with my plugin. Everything works fine, thanks for following up on this!

**Comment by mattwthompson** [comment]
Superseded by #651

**Pull request merged**